src/cargo/ops/cargo_new.rs

```diff
@@ -258,6 +258,13 @@ fn check_name(
             name
         ))?;
     }
+    if name != name.to_lowercase() {
+        shell.warn(format!(
+            "the name `{}` does not follow the preferred snake case format\n\
+            See https://github.com/rust-lang/rfcs/blob/master/text/0430-finalizing-naming-conventions.md#General-naming-conventions",
+            name
+        ))?;
+    }
```

Comment:
While this new style is simpler and allows for kebab case package names, the message makes it sound like a kebab case package name wouldn't be allowed.